EIPS/eip-5027.md

```diff
@@ -0,0 +1,84 @@
+
+The proposal implements a solution to remove the existing 24K limit of the code size.  Further, the proposal aims to minimize the changes in the client implementation (e.g., Geth) with
+- proper gas metering to avoid abusing the node resources for contract-related opcodes, i.e, `CODESIZE/CODECOPY/EXTCODESIZE/EXTCODECOPY/EXTCODEHASH/DELEGATECALL/CALL/CALLCODE/STATICCALL/CREATE/CREATE2`; and
+- no change to the existing structure of the Ethereum state.
+
+
+## Specification
+
+### Parameters
+
+| Constant                  | Value            |
+| ------------------------- | ---------------- |
+| `FORK_BLKNUM`             | TBD              |
+| `CODE_SIZE_UNIT`          | 24K              |
+| `READ_GAS_PER_UNIT`       | 700              |
+| `CREATE_DATA_GAS`         | 200              |
+
+If `block.number >= FORK_BLKNUM`, the contract creation initialization can return data with any length, but the contract-related opcodes will take extra gas as defined below:
+
+- For `CODESIZE/CODECOPY/EXTCODESIZE/EXTCODEHASH`, the gas is unchanged.
+
+- For `EXTCODECOPY/CALL/CALLCODE/DELEGATECALL/STATICCALL`, if the contract code size > `CODE_SIZE_UNIT`, then the opcodes will take extra gas as
+
+```
+(CODE_SIZE - 1) // CODE_SIZE_UNIT * READ_GAS_PER_UNIT
+```
+
+where `//` is the integer divide operator.
+
+- For CREATE/CREATE2, if the newly created contract size > `CODE_SIZE_UNIT`, the opcodes will take extra write gas as
+
+`(CODE_SIZE - CODE_SIZE_UNIT) * CREATE_DATA_GAS`.
+
+## Rationale
+
+### Gas Metering
+The goal is to measure the CPU/IO cost of the contract read/write operations reusing existing gas metering so that the resources will not be abused.
+
+- For code-size-related opcodes (`CODESIZE/EXTCODESIZE`), we would expect the client to implement a mapping from the hash of code to the size, so reading the code size of a large contract should not take O(1) resources.
+
+- For `CODECOPY`, the data is already loaded in memory (as part of `CALL/CALLCODE/DELEGATECALL/STATICCALL`), so we do not charge extra gas.
+
+- For `EXTCODEHASH`, the value is already in the account, so we do not charge extra gas.
+
+- For `EXTCODECOPY/CALL/CALLCODE/DELEGATECALL/STATICCALL`, since it will read extra data from the database, we will additionally charge READ_GAS_PER_UNIT per extra `CODE_SIZE`. Note that `READ_GAS_PER_UNIT = CALLGAS = EXTCODECOPYBASE = 700`.
+
+- For `CREATE/CREATE2`, since it will create extra data to the database, we will additionally charge `CREATE_DATA_GAS` per extra bytes.
+
+## Reference Implementation
+
+The reference implementation on Geth is available at https://github.com/QuarkChain/go-ethereum/pull/76.
```

Comment:
Hm... I don't think adding all of geth into this repository is feasible, but we're also supposed to discourage external links.

Maybe include a patch block with the changes? Seems fairly small.

Reply:
Maybe a link to the specific commit is better, as that will persist even if the PR and branch is deleted: 
https://github.com/QuarkChain/go-ethereum/tree/82ce60226a81e6a3acb92525482c16d254f63816

Reply:
Thanks for the comment.  I have replaced the link to the permalink as @sambacha suggested.

Reply:
Thanks for the comment. I have added the patch under the asset path. Please take a look.